pyomo/gdp/plugins/bigm.py

```diff
@@ -147,11 +149,21 @@ def _get_bigm_suffix_list(self, block):
         # SimpleBlocks. Though it is possible at this point to stick them
         # on whatever components you want, we won't pick them up.
         suffix_list = []
+        # first descend into the subblocks here to see if anything is there
+        for b in block.component_data_objects(Block, descend_into=(Block),
+                                              active=True,
+                                              sort=SortComponents.deterministic):
+            bigm = b.component('BigM')
+            if type(bigm) is Suffix:
+                suffix_list.append(bigm)
```

Comment:
Oh, oops, you're definitely right. But we still have a problem, which is that
```
m.b = Block()
m.b.d = Disjunct()
m.b.d.foo = Block()

m.b.BigM = Suffix()
m.b.BigM[None] = 10
m.b.d.foo.c = Constraint(...)
m.b.d.foo.BigM = Suffix()
m.b.d.foo.BigM[None] = 1
```
doesn't work with the original implementation, since `_get_bigm_suffix_list` only gets called on Disjuncts (we never looked on `foo`). But that's the real bug: We should be calling it on the parent block of every constraint...